### ISSUE TITLE
Used FeeMarketEIP1559Transaction and sign transaction for the Matic Mumbai testnet

### ISSUE THREAD

**Issue by NabiKAZ** [ethereumjs/ethereumjs-monorepo#1838]
I want to use `EIP1559` Transaction in the Matic `Mumbai` testnet. I have problem with sign that transaction.
I use ES6 and my file name have `.mjs` extension, so I was write this code:

```javascript
import {default as _common, Chain, Hardfork, CustomChain} from '@ethereumjs/common';
const Common = _common.default
import pkg from '@ethereumjs/tx'
const { FeeMarketEIP1559Transaction } = pkg;

const txParams = {
  nonce: '0x00',
  gasPrice: "0x"+(30 * 1000000000).toString(16),
  gasLimit: "0x"+(500000).toString(16),
  to: '0x0000000000000000000000000000000000000000',
  value: '0x00',
  data: '0x7f7465737432000000000000000000000000000000000000000000000000000000600057',
}

const common = Common.custom(CustomChain.MaticMumbai)

const tx = FeeMarketEIP1559Transaction.fromTxData(txParams, { common })

const privateKey = Buffer.from(
  SECRET,
  'hex',
)
const signedTx = tx.sign(privateKey)
const serializedTx = signedTx.serialize()
console.log('serializedTx', serializedTx.toString('hex'));
```

```
+-- @ethereumjs/common@2.6.3
+-- @ethereumjs/tx@3.5.1
+-- ethereumjs-tx@2.1.2
+-- web3@1.7.1

>node --version
v17.2.0
```

When I use
```javascript
const common = Common.custom(CustomChain.MaticMumbai)
```
Get this error:
```
Error: EIP-1559 not enabled on Common
```

When I use
```javascript
const common = new Common({ chain: CustomChain.PolygonMumbai, hardfork: Hardfork.London, eips: [1559] })
```
Get this error:
```
Error: Chain with name polygon-mumbai not supported
```

When I use
```javascript
const customChainParams = { name: 'matic-mumbai', chainId: 80001, networkId: 80001 }
const common = Common.forCustomChain('goerli', customChainParams)
```
Get this error:
```
Error: EIP-1559 not enabled on Common
```

But this for another network:
```javascript
const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.London })
```
Works well!

I don't have problem with `Transaction` function, Just there are problem with `FeeMarketEIP1559Transaction` for Matic Mumbai.
How can I do this and sign it? 😕

**Comment by cbrzn** [comment]
hey @NabiKAZ can you please check this link https://github.com/ethereumjs/ethereumjs-monorepo/issues/1836#issuecomment-1089011160 - this should solve your problem

**Comment by holgerd77** [comment]
Will close this for now, feel free to reopen if issue still persists.

**Comment by NabiKAZ** [comment]
Interestingly, the same problem was opened at the [same time](https://github.com/ethereumjs/ethereumjs-monorepo/issues/1836#issuecomment-1089011160)! 😄 

Thanks, the problem was solved with this:
```javascript
const common = Common.custom({ chainId: 80001 }, { hardfork: Hardfork.London })
```

Note that when using `FeeMarketEIP1559Transaction` the parameters `maxPriorityFeePerGas` and `maxFeePerGas` also seem required. Also, `gasLimit` should be used instead of the `gas` parameter.

I will put the full code of signing the transaction and sending it to the network below, it might be useful for someone:

```javascript
const ADDRESS = '......';
const SECRET = '......';
const API_KEY = '......';

import Web3 from 'web3';
import {default as _common, Chain, Hardfork, CustomChain} from '@ethereumjs/common';
const Common = _common.default
import pkg from '@ethereumjs/tx'
const { FeeMarketEIP1559Transaction } = pkg;

const rpcURL = 'https://rpc-mumbai.maticvigil.com/v1/' + API_KEY;
const web3 = new Web3(rpcURL);

const count = await web3.eth.getTransactionCount(ADDRESS);

const txParams = {
  nonce: web3.utils.toHex(count),
  gasPrice: web3.utils.toHex(web3.utils.toWei('30', 'gwei')),
  gasLimit: web3.utils.toHex(1000000),
  maxPriorityFeePerGas: web3.utils.toHex(web3.utils.toWei('3', 'gwei')),
  maxFeePerGas: web3.utils.toHex(web3.utils.toWei('8', 'gwei')),
  to: '0x0000000000000000000000000000000000000000',
  value: '0x00',
  data: '0x00',
}

const common = Common.custom({ chainId: 80001 }, { hardfork: Hardfork.London })

const tx = FeeMarketEIP1559Transaction.fromTxData(txParams, { common })

const privateKey = Buffer.from(
  SECRET,
  'hex',
)
const signedTx = tx.sign(privateKey)
const serializedTx = signedTx.serialize()
const serializedTxHex = web3.utils.toHex(serializedTx)

console.log('serializedTxHex', serializedTxHex);

web3.eth.sendSignedTransaction(serializedTxHex)
.on('transactionHash', function(hash){
	console.log('transactionHash', hash);
})
.on('receipt', function(receipt){
	console.log('receipt', receipt);
})
.on('confirmation', function(confirmationNumber, receipt){
	console.log('confirmation.confirmationNumber', confirmationNumber);
	console.log('confirmation.receipt', receipt);
})
.on('error', function(error, receipt){
	console.log(error, receipt);
});
```

enjoy! 😄

**Comment by NabiKAZ** [comment]
Now I have a question.
With `CustomChain.PolygonMumbai`, why do I have to manually configure `networkId: 80001` and `hardfork: Hardfork.London` in the input manually? Shouldn't something like this work automatically:
```javascript
const common = Common.custom(CustomChain.MaticMumbai)
```

**Comment by holgerd77** [comment]
> ```js
> const common = Common.custom(CustomChain.MaticMumbai)
> ```

Yes, partly, initialization like above will initialize with the correct chainID, for the `Common` library the `custom` method has no notion of hardforks though (there are no assiciated chain files), so you still have to set the HF manually.

This should be a complete example:

```typescript
import Common, { Hardfork, CustomChain } from '@ethereumjs/common'
const common = Common.custom(CustomChain.PolygonMumbai, { hardfork: Hardfork.London})
```

**Comment by NabiKAZ** [comment]
@holgerd77 But that command don't works. I try:

```javascript
Common.custom(CustomChain.PolygonMumbai, { hardfork: Hardfork.London })
//ERROR: EIP-1559 not enabled on Common

Common.custom({ chain: CustomChain.PolygonMumbai }, { hardfork: Hardfork.London })
//ERROR: Returned error: invalid sender

Common.custom({ chainId: CustomChain.PolygonMumbai }, { hardfork: Hardfork.London })
//ERROR: Invalid character
```

Just this works:
```javascript
Common.custom({ chainId: 80001 }, { hardfork: Hardfork.London })
```

**Comment by holgerd77** [comment]
First version is valid:

```typescript
Common.custom(CustomChain.PolygonMumbai, { hardfork: Hardfork.London })
//ERROR: EIP-1559 not enabled on Common
```

Oh, that's actually a (relatively severe) bug, just had a look into the code base, we just don't forward the options in the `Common.custom()` method at all. 😬 😜

Will directly open a PR, relatively easy to fix (but will take some time until release).

**Comment by holgerd77** [comment]
@NabiKAZ this should fix it: #1851
